sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1771,6 +1771,53 @@ def test_ocsvm_vs_sgdocsvm():
     assert corrcoef >= 0.9
 
 
+def test_sgd_oneclass_convergence():
+    # Check that the optimization does not end early and that the stopping criterion
+    # is working. Non-regression test for 30027
+    for nu in [0.1, 0.5, 0.9]:
+        # no need for large max_iter
+        model = SGDOneClassSVM(
+            nu=nu, max_iter=100, tol=1e-3, learning_rate="constant", eta0=1e-3
+        )
+        model.fit(iris.data)
+        # 6 is the minimal number of iterations that should be surpassed, after which
+        # the optimization can stop
+        assert model.n_iter_ > 6
+
+
+def test_sgd_oneclass_vs_linear_oneclass():
+    # Test convergence vs. liblinear `OneClassSVM` with kernel="linear"
+    for nu in [0.1, 0.5, 0.9]:
+        # allow enough iterations, small dataset
+        model = SGDOneClassSVM(
+            nu=nu, max_iter=20000, tol=None, learning_rate="constant", eta0=1e-3
+        )
+        model_ref = OneClassSVM(kernel="linear", nu=nu, tol=1e-6)  # reference model
+        model.fit(iris.data)
+        model_ref.fit(iris.data)
+
+        preds = model.predict(iris.data)
+        dec_fn = model.decision_function(iris.data)
+
+        preds_ref = model_ref.predict(iris.data)
+        dec_fn_ref = model_ref.decision_function(iris.data)
+
+        dec_fn_corr = np.corrcoef(dec_fn, dec_fn_ref)[0, 1]
+        preds_corr = np.corrcoef(preds, preds_ref)[0, 1]
+        # check weights and intercept concatenated together for correlation
+        coef_corr = np.corrcoef(
+            np.concatenate([model.coef_, -model.offset_]),
+            np.concatenate([model_ref.coef_.flatten(), model_ref.intercept_]),
+        )[0, 1]
```

Comment:
Do we really need to check correlations here? Can't we just check the preds and decision values for closeness directly?